### PR TITLE
Add option to deploy wazuh

### PR DESCRIPTION
**Pull request by Alex-Welsh**

### CODE REVIEW

**Comment by Alex-Welsh** [comment]
Matching SKC PR here

File: README.rst [hunk 1]
```diff
@@ -136,6 +140,10 @@ nodes. Both virtual machines and baremetal are supported, but the
 `controller_disk_size` and `compute_disk_size` must be set to 0 when using
 baremetal host. This will stop a block device being allocated.
 
+If `deploy_wazuh` is set to true, an infrastructure VM will be created that
+hosts the Wazuh manager. The wazuh deployment playbooks will also be triggered
+automatically to deploy wazuh agents to the overcloud hosts.
```
Comment (MoteHue) [inline README.rst:145]:
```suggestion
If `deploy_wazuh` is set to true, an infrastructure VM will be created that
hosts the Wazuh manager. The Wazuh deployment playbooks will also be triggered
automatically to deploy Wazuh agents to the overcloud hosts.
```

File: variables.tf [hunk 2]
```diff
@@ -86,3 +90,15 @@ variable "storage_disk_size" {
...
 }
+
+variable "infra_vm_disk_size" {
+  description = "Block storage root disk size for infrastructure VMs."
+  type = number
+  default = 100
+}
+
+variable "deploy_wazuh" {
+  description = "Bool, whether or not to deploy wazuh"
```
Comment (MoteHue) [inline variables.tf:101]:
```suggestion
  description = "Bool, whether or not to deploy Wazuh."
```

**Review by MoteHue — CHANGES_REQUESTED** [review]

**Comment by Alex-Welsh** [comment]
Regarding Wazuh vs wazuh (capitalisation), I'm not sure which is correct. I was following what I found on their website: https://wazuh.com/ but I've since found contradicting examples

**Comment by MoteHue** [comment]
> Regarding Wazuh vs wazuh (capitalisation), I'm not sure which is correct. I was following what I found on their website: https://wazuh.com/ but I've since found contradicting examples

I'm seeing it capitalised on their website, for example throughout their quickstart guide: https://documentation.wazuh.com/current/quickstart.html

**Comment by Alex-Welsh** [comment]
> > Regarding Wazuh vs wazuh (capitalisation), I'm not sure which is correct. I was following what I found on their website: https://wazuh.com/ but I've since found contradicting examples
> 
> I'm seeing it capitalised on their website, for example throughout their quickstart guide: https://documentation.wazuh.com/current/quickstart.html

By `on their website` what I actually mean is I made it as far as their logo in the top left and though that's probably good enough

**Review by MoteHue — APPROVED** [review]

File: .terraform.lock.hcl [hunk 3]
```diff
@@ -1,43 +0,0 @@
-# This file is maintained automatically by "terraform init".
```
Comment (markgoddard) [inline .terraform.lock.hcl:1]: Why remove this?

Comment (Alex-Welsh) [reply .terraform.lock.hcl]: I misunderstood the file and thought it was unnecessary, I've re-added it back in

**Review by markgoddard — APPROVED** [review]

**Pull request merged**